### DISCUSSION TITLE
Accesso API

### DISCUSSION THREAD

**Discussion by martinapavonidoc1** [iisgiua/giuaschool#675] [Richieste di aiuto] [answered]
Buonasera , volevo chiedere come funzionasse l’accesso con google e come attivarlo con api per il registro elettronico. grazie

**Comment by trinko** [comment, marked as answer]
Premessa: non è molto semplice...

Devi avere già attivato il servizio Google Workspace for Education.

Sul registro vanno configurati i parametri (**SISTEMA > PARAMETRI**):
  -  **id_provider**: con il valore predefinito "gsuite"
  -  **id_provider_dominio**: con il dominio utilizzato, es. nomescuola.edu.it
  -  **id_provider_tipo**: con i codici per gli utenti che dovranno usare l'autenticazione, ad. es. "ADSP" indica gli alunni, i docenti, lo staff e il preside.


Poi va fatta la configurazione sulla console di amministrazione del tuo Google Workspace: segui le loro guide, come ad esempio [questa](https://developers.google.com/identity/protocols/oauth2/openid-connect#registeringyourapp)

Poi vanno inseriti i codici generati da Google nel file .env
```
OAUTH_GOOGLE_CLIENT_ID=...
OAUTH_GOOGLE_CLIENT_SECRET=...
OAUTH_GOOGLE_CLIENT_HD=...
```
